src/api/schema/log_events/mod.rs

```diff
@@ -0,0 +1,145 @@
+mod event;
+mod notification;
+
+use event::LogEvent;
+
+use crate::api::{
+    tap::{TapController, TapNotification, TapResult, TapSink},
+    ControlSender,
+};
+use async_graphql::{validators::IntRange, Context, Subscription, Union};
+use futures::StreamExt;
+use itertools::Itertools;
+use tokio::{select, stream::Stream, sync::mpsc, time};
+
+#[derive(Union)]
+/// Log event result which can be a log event or notification
+pub enum LogEventResult {
```

Comment:
This doesn't feel like a `Result`, since there's no error variant. Maybe something like `TapPayload` or similar?

Reply:
Yeah, this is an artefact of an earlier version where we had a `LogEventError` type, and it was deemed a subscription-canceling error for a component to go away after configuration had changed.

We then decided that a subscription should persist over reloads, irrespective of configuration, to allow for the use-case of observing a component's first events as it's added to config for the first time. LogEventError became `LogEventNotification`.

I'll change this to the `Payload` suffix, which better captures the new contents.